lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp

```diff
@@ -163,17 +192,36 @@ static LLVM::LLVMType getProcPersistenceTy(LLVM::LLVMDialect *dialect,
                                            ProcOp &proc) {
   SmallVector<LLVM::LLVMType, 3> types = SmallVector<LLVM::LLVMType, 3>();
   proc.walk([&](Operation *op) -> void {
-    if (op->isUsedOutsideOfBlock(op->getBlock())) {
+    if (op->isUsedOutsideOfBlock(op->getBlock()) || isWaitDestArg(op)) {
       if (auto ptr = op->getResult(0).getType().dyn_cast<PtrType>()) {
         // Persist the unwrapped value.
         auto converted = converter.convertType(ptr.getUnderlyingType());
         types.push_back(converted.cast<LLVM::LLVMType>());
+      } else if (op->getResult(0).getType().isa<SigType>()) {
+        // Persist unwrapped signal structure.
+        types.push_back(getSigType(dialect));
```

Comment:
I'm not sure how this works, but is something like:

`else if (auto ty = ...dyn_cast<SigType>()) { types.push_back(ty); }`

correct?

Reply:
Yes, `getSigType` returns the type I'm using to represent signals in LLVM dialect (more a pointer to a struct containing the information needed to work on signals, rather than just an opaque pointer). That helper name could be a bit misleading though, I refactored it to `getLLVMSigType`. Also I could actually squash this case with the pointer case above.

Reply:
LGTM!